rs/tests/research/spec_compliance/spec_compliance.rs

```diff
@@ -364,20 +322,22 @@ pub fn with_endpoint(
     env: TestEnv,
     test_subnet: SubnetSnapshot,
     peer_subnet: SubnetSnapshot,
-    use_bn: bool,
+    use_api_bn: bool,
     httpbin_proto: Option<String>,
     httpbin: Option<String>,
     log: Logger,
     excluded_tests: Vec<&str>,
     included_tests: Vec<&str>,
 ) {
-    let endpoint = if use_bn {
-        let boundary_node = env
-            .get_deployed_boundary_node(BOUNDARY_NODE_NAME)
-            .unwrap()
-            .get_snapshot()
-            .unwrap();
-        boundary_node.get_public_url().to_string()
+    let endpoint = if use_api_bn {
+        let api_bn = env
+            .topology_snapshot()
+            .api_boundary_nodes()
+            .next()
+            .expect("No API boundary node found");
+
+        let api_bn_addr = format!("[{}]", api_bn.get_public_addr().ip());
+        format!("https://{}/", api_bn_addr)
```

Comment:
Why can't we use `api_bn.get_public_url().to_string()`?

Reply:
Because there are no DNS entries for the public URL. In that case, we would have to create an HTTP client that internally resolves the domain name to the API BNs IP address. The API BNs however don't care about the host and just serve the traffic that arrives on port 443. Hence, we can just the IP address.

Reply:
> Hence, we can just the IP address.

How does https work with an IP address? Does the API BN have a self-signed certificate for its IP address?

> We could just change the get_public_url() of the API BN to return the IP address instead of its fake domain name.

That would make more sense in my opinion.

Reply:
Yes, the API BNs have a self-signed certificate. The agents used in the other tests just accept anything. I didn't look into your Haskell test, but since it worked, I assumed that your HTTP client also doesn't check the certificate.

Reply:
> I assumed that your HTTP client also doesn't check the certificate.

Correct, there's an option to allow self-signed certificates. I was mainly asking to confirm my understanding.

Reply:
Changed it as discussed.